v2/opttoy.go

```diff
@@ -528,6 +559,21 @@ func (e *expr) removeFilters() {
 	e.children = e.children[:int(filterStart)]
 }
 
+// attrs returns the attribute indexes in the order they are output by the
+// expression. The attribute indexes are computed from the output vars of the
+// projections.
+func (e *expr) attrs() []bitmapIndex {
```

Comment:
@knz The naming here is sub-par, but I think this gives us enough information to avoid needing to maintain a `reorder` array. Still TODO for me is to show how this would work for your `UNION ALL` query.

Reply:
I still do not understand why you need separate "projections" arrays and therefore I can't comment on whether this is useful.

Reply:
If `expr` doesn't contain projections, where else would they get stored?